lib/service_check.pm

```diff
@@ -247,7 +247,10 @@ sub install_services {
                     $service_type = 'Systemd';
                 }
                 if (exists $service->{$s}->{service_check_func}) {
-                    $service->{$s}->{service_check_func}->('before', $service_type);
+                    my %hash = $service->{$s};
+                    $hash{service_type} = $service_type;
+                    $hash{stage}        = 'before';
+                    $service->{$s}->{service_check_func}->(%hash);
```

Comment:
I think this should be rewritten to use hash parameters directly:

```suggestion
                    $service->{$s}->{service_check_func}->({$service->{$s}, service_type => $service_type, stage => 'before'});
```

Not entirely sure about syntax correctness though :)

Reply:
changed, tanks.